### PR TITLE
Fix reading large SequenceFile, RCFile, or Avro files

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Fix reading large SequenceFile, RCFile, or Avro files. ({issue}`18837`)
```

Fixes https://github.com/trinodb/trino/issues/17890

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

**Comment by wendigo** [comment]
@electrum can we have a test for it?

**Comment by electrum** [comment]
I'm not sure how to test this other than generating a 2GB+ file, which would negatively impact the CI, and doesn't seem to be worth the cost.

**Review by raunaqmorarka — APPROVED** [review]

**Review by dain — APPROVED** [review]

File: lib/trino-hive-formats/src/main/java/io/trino/hive/formats/TrinoDataInputStream.java [hunk 1]
```diff
@@ -230,10 +232,14 @@ public int read()
     public long skip(long length)
             throws IOException
     {
+        if (length <= 0) {
+            return 0;
```
Comment (findepi) [inline lib/trino-hive-formats/src/main/java/io/trino/hive/formats/TrinoDataInputStream.java:236]: for length == 0, it's obvious.
why also for length < 0? is it worth explanatory comment?

Comment (electrum) [reply lib/trino-hive-formats/src/main/java/io/trino/hive/formats/TrinoDataInputStream.java:236]: This is the behavior specified in the Javadoc and in the default implementation.

**Pull request merged**